File: MyBudget-back/src/main/java/com/example/mybudget/services/TransactionService.java
package com.example.mybudget.services;

import com.example.mybudget.models.dto.Transaction;
import com.example.mybudget.models.requests.TransactionRequest;

import java.util.List;

public interface TransactionService {
    Transaction create(TransactionRequest request);

    List<Transaction> getAll();

    List<Transaction> getByAccount(Integer accountId);
}

File: MyBudget-front/src/app/services/transaction.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Transaction } from '../models/transaction';
import { TransactionRequest } from '../models/transaction-request';

@Injectable({
  providedIn: 'root',
})
export class TransactionService {
  private apiUrl = 'http://localhost:8080/transactions';

  constructor(private http: HttpClient) {}

  getAll() {
    return this.http.get<any>(this.apiUrl);
  }

  getByAccount(accountId: number): Observable<Transaction[]> {
    return this.http.get<Transaction[]>(`${this.apiUrl}/${accountId}`);
  }

  create(request: TransactionRequest) {
    return this.http.post<Transaction>(this.apiUrl, request);
  }
}
